cmd/kubeadm/app/features/features.go

```diff
@@ -148,7 +149,7 @@ func NewFeatureGate(f *FeatureList, value string) (map[string]bool, error) {
 		}
 
 		if featureSpec.PreRelease == featuregate.Deprecated {
-			return nil, errors.Errorf("feature-gate key is deprecated: %s", k)
+			klog.Warningf("Setting deprecated feature gate %s=%t. It will be removed in a future release.", k, v)
```

Comment:
i think @SataQiu pointed out we also error during upgrade. we might want to change that too.

Reply:
https://github.com/kubernetes/kubernetes/blob/e5e354c73d06cf9071cd165bcf654f37fc437910/cmd/kubeadm/app/features/features.go#L150-L152

Here needs an update as well.